src/test/java/com/coveo/spillway/SpillwayTryUpdateAndVerifyLimitTest.java

```diff
@@ -223,7 +226,7 @@ public void capacityNotIncrementedIfLimitTriggered() throws Exception {
 
     Map<LimitKey, Integer> counters = inMemoryStorage.getCurrentLimitCounters();
 
-    assertThat(counters.values()).containsExactly(1);
+    assertThat(counters.values()).containsExactly(0);
```

Comment:
This is pretty much something changed/broke by the last PR we merged. The original logic was as it is. In this, I have reverted back to original. 

To explain this,
This test creates a counter with limit = 0. So we try to count 1 event (which exceeds the limit) and then verify that the internal counter is also zero (This is the logic of internal counter would not count beyond it's limit value).

However, with the changes of last PR, we first increment, then compare and if it pass the limit, we throw exceed limit. But with that, we ended up incrementing the counter by one so the internal counters final value becomes limit + 1. 

So the code was changed as: 

In the storage level where counter is calculated:

`public Integer addAndGetWithLimit(int cost, int limit) {
    return delta.accumulateAndGet(cost, (left, right) -> left > limit ? left : left + right)
        + total.get();
  }`

accumulateAndGet increment the current counter with cost (+1) and return the value. Which we compare here in Spillway::updateAndVerifyExceededLimits

`if (result.getValue() > limit.getCapacity(context)) {
          exceededLimits.add(limit.getDefinition());
        }`

While working on the sliding window concepts, I realized it's needed to enforce the same logic and it was before and it can be done by `AtomicInteger::getAndAccumulate `  This will accumulate the value but returns the past value than the new value.

So the improved code is:

`public Integer addAndGetWithLimit(int cost, int limit, int previousCount) {
    return delta.getAndAccumulate(
            cost,
            (left, right)
                -> right + left + previousCount + total.get() > limit ? left : left + right)
        + total.get();
  }` 

The previousCount above is used for sliding window calculation. The main difference is getAndAccumulate vs. accumulateAndGet.

So now, in the Spillway level, we do:

`int counterValue = result.getValue() + cost;
        handleTriggers(context, cost, now, counterValue, limit);
        if (counterValue > limit.getCapacity(context)) {
          exceededLimits.add(limit.getDefinition());
        }`

So we now compare without need to exceed the counter value, which revert back to the original logic.
Hope this is clear :)